src/pyclass.rs

```diff
@@ -137,18 +137,14 @@ where
     // type size
     type_object.tp_basicsize = std::mem::size_of::<T::Layout>() as ffi::Py_ssize_t;
 
-    let mut offset = type_object.tp_basicsize;
-
     // __dict__ support
-    if let Some(dict_offset) = T::Dict::OFFSET {
-        offset += dict_offset as ffi::Py_ssize_t;
-        type_object.tp_dictoffset = offset;
+    if !T::Dict::IS_DUMMY {
```

Comment:
How about moving this if statement to the `PyCell`?
I don't like that `dict_offset` can return a meaningless value even if it doesn't have a dict.

Reply:
Good catch, I `PyCell::*_offset` now returns `Option<usize>`